crates/bindings-csharp/Runtime/bindings.c

```diff
@@ -1,781 +1,231 @@
-                                                    int64_t arg1,
-                                                    int64_t arg2,
-                                                    int32_t arg3) {
-  return 0;
-}
-
-int32_t __imported_wasi_snapshot_preview1_fd_write(int32_t arg0,
-                                                   int32_t arg1,
-                                                   int32_t arg2,
-                                                   int32_t arg3) {
-  return 0;
-}
-
-int32_t __imported_wasi_snapshot_preview1_fd_allocate(int32_t arg0,
-                                                      int64_t arg1,
-                                                      int64_t arg2) {
-  return 0;
-}
-
-int32_t __imported_wasi_snapshot_preview1_fd_close(int32_t arg0) {
-  return 0;
-}
-
-int32_t __imported_wasi_snapshot_preview1_fd_datasync(int32_t arg0) {
-  return 0;
-}
-
-int32_t __imported_wasi_snapshot_preview1_fd_fdstat_get(int32_t arg0,
-                                                        int32_t arg1) {
-  return 0;
-}
-
-int32_t __imported_wasi_snapshot_preview1_fd_fdstat_set_flags(int32_t arg0,
-                                                              int32_t arg1) {
-  return 0;
-}
-
-int32_t __imported_wasi_snapshot_preview1_fd_fdstat_set_rights(int32_t arg0,
-                                                               int64_t arg1,
-                                                               int64_t arg2) {
-  return 0;
-}
-
-int32_t __imported_wasi_snapshot_preview1_fd_filestat_get(int32_t arg0,
-                                                          int32_t arg1) {
-  return 0;
-}
-
-int32_t __imported_wasi_snapshot_preview1_fd_filestat_set_size(int32_t arg0,
-                                                               int64_t arg1) {
-  return 0;
-}
-
-int32_t __imported_wasi_snapshot_preview1_fd_filestat_set_times(int32_t arg0,
-                                                                int64_t arg1,
-                                                                int64_t arg2,
-                                                                int32_t arg3) {
-  return 0;
-}
-
-int32_t __imported_wasi_snapshot_preview1_fd_pread(int32_t arg0,
-                                                   int32_t arg1,
-                                                   int32_t arg2,
-                                                   int64_t arg3,
-                                                   int32_t arg4) {
+// For `fd_write`, we need to at least collect and report sum of sizes.
+// If we report size 0, the caller will assume that the write failed and will
+// try again, which will result in an infinite loop.
+int32_t WASI_NAME(fd_write)(__wasi_fd_t fd, const __wasi_ciovec_t* iovs,
+                            size_t iovs_len, __wasi_size_t* retptr0) {
+  for (size_t i = 0; i < iovs_len; i++) {
+    // Note: this will produce ugly broken output, but there's not much we can

```

Comment:
Maybe a p2 ticket tracking this?

Reply:
We already have one + implementation in a branch I should complete, but I've been waiting for migration to Wasmtime to happen first as it's much easier to do custom WASI on top of it.